packages/blocks/src/page-block/utils/bind-hotkey.ts

```diff
@@ -103,25 +110,45 @@ export function bindCommonHotkey(page: Page) {
   // We shouldn't prevent user input, because there could have CN/JP/KR... input,
   // that have pop-up for selecting local characters.
   // So we could just hook on the keydown event and detect whether user input a new character.
-  hotkey.addListener(HOTKEYS.ANY_KEY, e => {
-    if (!isPrintableKeyEvent(e) || page.readonly) return;
-    const blockRange = getCurrentBlockRange(page);
-    if (!blockRange) {
-      return;
-    }
-    if (blockRange.type === 'Block') {
-      handleKeydownAfterSelectBlocks({
-        page,
-        keyboardEvent: e,
-        selectedBlocks: blockRange.models,
-      });
-      return;
-    }
+  hotkey.addListener(
+    HOTKEYS.ANY_KEY,
+    e => {
+      if (anyKeyOptions) {
+        if (anyKeyOptions.pressed === e.key && e.type === 'keyup') {
+          anyKeyOptions.keyup(e);
+          anyKeyOptions.pressed = undefined;
+          return;
+        }
+        if (anyKeyOptions.filter(e)) {
+          anyKeyOptions.pressed = e.key;
+          anyKeyOptions.keydown(e);
+          return;
+        }
+      }
 
-    const range = blockRangeToNativeRange(blockRange);
-    if (!range || !isMultiBlockRange(range)) return;
-    deleteModelsByRange(page);
-  });
+      if (e.type === 'keyup') return;
```

Comment:
The logic used in this section is confusing. Therefore, it would be better to separate `HOTKEYS.ANY_KEY` from `bindCommonHotkey` and process it independently